spec/models/manageiq/providers/ibm_cloud/power_virtual_servers/storage_manager/cloud_volume_spec.rb

```diff
@@ -0,0 +1,81 @@
+describe ManageIQ::Providers::IbmCloud::PowerVirtualServers::StorageManager::CloudVolume do
+  let(:ems_cloud) { FactoryBot.create(:ems_ibm_cloud_power_virtual_servers_cloud) }
+  let(:ems_storage) { FactoryBot.create(:ems_ibm_cloud_power_virtual_servers_storage, :parent_ems_id => ems_cloud.id) }
+  let(:cloud_volume) { FactoryBot.create(:cloud_volume_ibm_cloud_power_virtual_servers, :ext_management_system => ems_storage) }
+
+  describe "#validate_delete_volume" do
+    it "validates volume delete when in-use" do
+      cloud_volume.update(:status => "in-use")
```

Comment:
Usually with rspec we'll do something like this:
```
let(:cloud_volume) { FactoryBot.create(..., :status => status) }

describe "#validate_delete_volume" do
  context "when available" do
    let(:status) { "available" }
  end

  context "when unavailable" do
    let(:status) { "unavailable" }
  end
end
```

instead of `cloud_volume.update(:status)` since cloud_volume is actually created every `it` you're basically doing `FactoryBot.create().update()` here

Reply:
Done, and I finally understand how `let()` is used. There is some clutter from contexts with only one test. Would it be better to try to clean that up, or to keep a uniform format across all tests?

Reply:
One thing you could do is set some defaults at the top level like multi-attachment false so you don't have to set it in every context if it isn't relevant to the context you're testing